packages/front-end/components/Settings/SecretApiKeys.tsx

```diff
@@ -14,103 +15,202 @@ const SecretApiKeys: FC<{ keys: ApiKeyInterface[]; mutate: () => void }> = ({
-        </button>
-      )}
+
+      {/* region Secret API keys */}
+      <div className="mb-5">
+        <h1>Secret API Keys</h1>
+        <p>
+          Secret keys have full read and write access to your organization.
+          Because of this, they must be kept secure and{" "}
+          <strong>must not be exposed to users</strong>.
+        </p>
+        {secretKeys.length > 0 && (
+          <ApiKeysTable
+            onDelete={onDelete}
+            keys={secretKeys}
+            canManageKeys={canManageKeys}
+            onReveal={onReveal}
+          />
+        )}
+        {canManageKeys && (
+          <button
+            className="btn btn-primary"
+            onClick={(e) => {
+              e.preventDefault();
+              // Secret API keys have no type
+              setModalApiKeyType(undefined);
+              setOpen(true);
+            }}
+          >
+            <FaKey /> Create New Secret Key
+          </button>
+        )}
+      </div>
+      {/* endregion Secret API keys */}
+
+      {/* region User API keys */}
+      <div className="mb-5">
+        <h1>User API Keys</h1>
+        <p>
+          User keys have full read and write access to your user account.
+          Because of this, they must be kept secure and{" "}
+          <strong>must not be exposed to users</strong>.
+        </p>
+        {userKeys.length > 0 && (
+          <ApiKeysTable
+            onDelete={onDelete}
+            keys={userKeys}
+            canManageKeys={canManageKeys}
+            onReveal={onReveal}
+          />
+        )}
+        {canManageKeys && (
+          <button
+            className="btn btn-primary"
+            onClick={(e) => {
+              e.preventDefault();
+              setModalApiKeyType("user");
+              setOpen(true);
+            }}
+          >
+            <FaKey /> Create New User Key
+          </button>
+        )}
+      </div>
+      {/* endregion User API keys */}
+
+      {/* region Read-only API keys */}
+      <div className="mb-5">
+        <h1>Read-only API Keys</h1>
+        <p>Read-only API keys have read access to resources.</p>
+        {readOnlyKeys.length > 0 && (
+          <ApiKeysTable
+            onDelete={onDelete}
+            keys={readOnlyKeys}
+            canManageKeys={true}
+            onReveal={onReveal}
+          />
+        )}
+        {canManageKeys && (
+          <button
+            className="btn btn-primary"
+            onClick={(e) => {
+              e.preventDefault();
+              setModalApiKeyType("read-only");
+              setOpen(true);
+            }}
+          >
+            <FaKey /> Create New Read-only Key
+          </button>
+        )}
+      </div>
+      {/* endregion Read-only API keys */}
     </div>
   );
 };
 
+const ApiKeysTable: FC<{
+  onDelete: (keyId: string | undefined) => () => Promise<void>;
+  keys: ApiKeyInterface[];
+  canManageKeys: boolean;
+  onReveal: (keyId: string | undefined) => () => Promise<string>;
```

Comment:
consider a `maxHeight` and `overflow: auto` since we're stacking these vertically and each table could potentially get quite long